src/App.tsx

```diff
@@ -33,7 +33,28 @@ import {
 import { colorSchemeAtom } from './common/colors';
 import { useCurrentUser } from './common/hooks/useCurrentUser';
 import { useRefetch } from './common/hooks/useRefetch';
+import { PreventNavigationModal } from './components/PreventNavigationModal';
+import { isNavigationModalVisibleAtom } from './common/hooks/usePreventNavigation';
 
+interface PreventLeavingPage {
+  prevent: boolean;
+  actionKey?: 'switchCompany' | 'browserBack';
+}
+
+interface HistoryLocation {
+  lastLocation: string;
+  nonPreventedLocations: string[];
+}
+
+export const preventLeavingPageAtom = atom<PreventLeavingPage>({
+  prevent: false,
+  actionKey: undefined,
+});
+
+export const lastHistoryLocationAtom = atom<HistoryLocation>({
+  lastLocation: '',
+  nonPreventedLocations: [],
+});
```

Comment:
app.tsx is getting quite chunky, please move this into a separate module or usePreventNavigation, that works, too.

Reply:
@beganovich Yeah, I separated it into the hook called `useAddPreventNavigationEvents`.